vexcl/optimisation.hpp

```diff
@@ -0,0 +1,200 @@
+#ifndef OPTIMISATIONS_HPP
```

Comment:
I don't think that 'optimization' is a good name for this functionality. What about 'qualifiers' or even 'pointer_qualifiers'? Anyway, to make it consistent with the rest of the code, the include guard should contain the full path to the file from vexcl root folder. (as in `VEXCL_OPTIMIZATIONS_HPP`).